packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -770,7 +797,7 @@ class _RenderSegmentedButton<T> extends RenderBox with
 
   Size _calculateChildSize(BoxConstraints constraints) {
     double maxHeight = 0;
-    double childWidth = constraints.minWidth / childCount;
+    double childWidth = _isExpanded ? constraints.maxWidth : constraints.minWidth / childCount;
     RenderBox? child = firstChild;
     while (child != null) {
       childWidth = math.max(childWidth, child.getMaxIntrinsicWidth(double.infinity));
```

Comment:
Since `_isExpanded` essentially hard-codes the width of a child to `constraints.maxWidth / childCount` you should be able to skip the getMaxIntrinsicWidth altogether making this calculation more efficient.

Reply:
You are so right, I missed this. l fixed it now